src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -76,8 +77,12 @@ public TaskData getTaskByID(long taskID) {
   }
 
   public void deleteTasks(ArrayList<Long> taskIDs) {
-    ArrayList<Key> keys = getKeysFromTaskIDs(taskIDs);
-    datastore.delete(keys);
+    if (taskIDs.isEmpty()) {
+      return;
+    }
```

Comment:
Interesting approach with the recursive deletes.  So this calls getTasks recursively, then calls delete for each returned set.  One possible danger here would be if one of the deletes failed, you'd then be left with orphan records (since the parent task no longer exists).

A couple alternatives you can look  into:
1.  Still do recursive getTasks but do one giant delete at the end
2.  Look into enabling "transactions"